### PR TITLE
[SPARK-35427][SQL][TESTS] Check the `EXCEPTION` rebase mode for Avro/Parquet

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
Add tests to check the `EXCEPTION` rebase mode explicitly in the datasources:
- Parquet: `DATE` type and `TIMESTAMP`: `INT96`, `TIMESTAMP_MICROS`, `TIMESTAMP_MILLIS`
- Avro: `DATE` type and `TIMESTAMP`: `timestamp-millis` and `timestamp-micros`.

### Why are the changes needed?
1. To improve test coverage
2. The `EXCEPTION` rebase mode should be checked independently from the default settings.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running the affected test suites:
```
$ build/sbt "test:testOnly *AvroV2Suite"
$ build/sbt "test:testOnly *ParquetRebaseDatetimeV1Suite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan Could you review this PR, please.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala [hunk 1]
```diff
@@ -374,6 +375,58 @@ abstract class ParquetRebaseDatetimeSuite
       }
     }
   }
+
+  test("SPARK-35427: datetime rebasing in the EXCEPTION mode") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala:379]: do we need to clean up the existing tests to remove code that are being tested here?

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala:379]: Let me find out the places where we check the default mode (`EXCEPTION`).

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala:379]: I think we can avoid checking the default `EXCEPTION` mode at least in the places:
1. https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala#L142-L146
2. https://github.com/apache/spark/blob/2bd32548f520ad3fc119e54ee3b4b4f27d33e046/external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala#L1835-L1837

@cloud-fan Let me know if you think that there are other places.

**Comment by MaxGekk** [comment]
@cloud-fan Any objections to the changes?

**Comment by MaxGekk** [comment]
@gengliangwang @HyukjinKwon Could you take a look at the PR, please.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**